language/go/modules.go

```diff
@@ -40,22 +40,35 @@ func importReposFromModules(args language.ImportReposArgs) language.ImportReposR
 
 	// List all modules except for the main module, including implicit indirect
 	// dependencies.
-	type module struct {
+	// Schema is documented at https://go.dev/ref/mod#go-list-m
+	type moduleFromList struct {
 		Path, Version, Sum, Error string
 		Main                      bool
 		Replace                   *struct {
 			Path, Version string
 		}
 	}
 	// path@version can be used as a unique identifier for looking up sums
-	pathToModule := map[string]*module{}
+	pathToModule := map[string]*moduleFromList{}
 	data, err := goListModules(dir)
+	dec := json.NewDecoder(bytes.NewReader(data))
 	if err != nil {
+		// Best-effort try to adorn specific error details from the JSON output.
+		for dec.More() {
+			var dl moduleFromList
+			if err := dec.Decode(&dl); err != nil {
+				// If we couldn't parse a possible error description, just return the raw error.
+				break
```

Comment:
Nit: could we wrap the error before breaking to let user know that this was caused specifically when we try to decode `go list` json output?